logstash-core/lib/logstash/modules/importer.rb

```diff
@@ -0,0 +1,39 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging"
+
+module LogStash module Modules class Importer
+  include LogStash::Util::Loggable
+
+  attr_reader :client
+
+  def initialize(client)
+    @client = client
+  end
+
+  def put(resource, overwrite = true)
+    path = resource.import_path
+    logger.info("Attempting PUT", :url_path => path, :file_path => resource.content_path)
+    if !overwrite && content_exists?(path)
+      logger.debug("Found existing Elasticsearch resource.", :resource => path)
+      return
+    end
+    put_overwrite(path, resource.content)
+  end
+
+  private
+
+  def put_overwrite(path, content)
+    if content_exists?(path)
+      response = @client.delete(path)
+    end
+    # hmmm, versioning?
```

Comment:
By this I mean that if the content has a version, in the future, do we want to check that we are not overwriting a newer version with an older one. Quite a general question based on the content and the content type.